torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -192,6 +203,10 @@ class GlooStore : public ::gloo::rendezvous::Store {
     store_->wait(keys, timeout);
   }
 
+  c10::intrusive_ptr<::c10d::Store> getStore() {
```

Comment:
This `getStore` is exposed on `ProcessGroupGloo::GlooStore`, but you actually want to expose it on `ProcessGroupGloo` and return a `GlooStore`, I think that is why we have the compiler error.

As for a default, I think `store` is always set by constructor so we don't have to worry about null etc if we make the above change.